sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidSqlValidator.java

```diff
@@ -39,4 +44,23 @@ protected DruidSqlValidator(
   {
     super(opTab, catalogReader, typeFactory, validatorConfig);
   }
+
+  @Override
+  public void validateCall(SqlCall call, SqlValidatorScope scope)
+  {
+    if (call.getKind() == SqlKind.NULLS_FIRST) {
+      SqlNode op0 = call.getOperandList().get(0);
+      if (op0.getKind() == SqlKind.DESCENDING) {
+        throw InvalidSqlInput.exception("DESCENDING ordering with NULLS FIRST is not supported!");
```

Comment:
I've had experiences with really large SQL queries where a message like this tells me what's wrong, but doesn't always inform me of where I made the mistake.  When we convert a `ValidationException` from Calcite into an `InvalidSqlInput` exception, it comes with the location of the SQL issue, which is helpful context.  I wonder if either

a) We can add more context here to the error message to help identify which part of a potentially huge, sprawling SQL statement has an issue, OR
b) Maybe in the `validateCall()` method it's correct to throw a ValidationException to get the Calcite system to add the context of where it happened and then rely on the normal conversion of that to the `InvalidSqlInput` exception on the way out?

Reply:
initially I've used use validationError....but then I've seen a lot of place where `InvalidSqlInput` was used in similar situations like [here in the DruidSqlParserUtils](https://github.com/apache/druid/blame/a8febd457ceb184870dfbb11f6fe82b02dd1242d/sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlParserUtils.java#L247); and I was just "following the current practices"

Reply:
I think it would probably make more sense to leave it like this for now -  and instead of just fixing this one instance - replace all possible `InvalidSqlInput` -s with an alternate (which could pinpoint the sql location) and use that everywhere.

Reply:
I think there are some places where `InvalidSqlInput` can't really point to a sql location. I agree with the fact that we can take that up as a whole, but I agree with Eric that we should provide some context info in the error message. If we are to go with the `InvalidSqlInput` way, we would add context to it like https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java#L421
What do you think?

Reply:
yeah; I guessed there will be some which can't be changed - however at a lot of callsites we do have an `SqlNode` ; I had in mind to target those